src/app/services/wallet.service.ts

```diff
@@ -24,94 +24,134 @@ export class WalletService {
     );
 
-    this.syncWallet(selectedWalletId, passphrase).subscribe(data => {
-      if (_.isUndefined(data["result"])) {
-        result.next(false);
-      } else {
-        this.checkWalletBalance(selectedWalletId, passphrase).subscribe(data => {
-          if (_.isNil(data["result"])) {
-            result.next(false);
-          } else {
-            const balance = new BigNumber(data["result"])
-              .dividedBy("100000000")
-              .toString(10);
-            console.log(data["result"]);
-            console.log(balance);
-            this.setWalletBalance(balance);
-            this.setDecryptedFlag(true);
-            result.next(true);
-            this.checkWalletAddress(selectedWalletId, passphrase).subscribe(
-              data => {
+    this.syncWallet(selectedWalletId, passphrase, walletEnckey).subscribe(
+      (data) => {
+        if (_.isUndefined(data["result"])) {
+          result.next(false);
+        } else {
+          this.checkWalletBalance(
+            selectedWalletId,
+            passphrase,
+            walletEnckey
+          ).subscribe((data) => {
+            if (_.isNil(data["result"])) {
+              result.next(false);
+            } else {
+              const balance = new BigNumber(data["result"]["total"])
+                .dividedBy("100000000")
+                .toString(10);
+              this.setWalletBalance(balance);
+              this.setDecryptedFlag(true);
+              result.next(true);
+              this.checkWalletAddress(
+                selectedWalletId,
+                passphrase,
+                walletEnckey
+              ).subscribe((data) => {
                 this.setWalletAddress(data["result"][0]);
-              }
-            );
-            this.checkWalletViewKey(selectedWalletId, passphrase).subscribe(
-              data => {
+              });
+              this.checkWalletViewKey(
+                selectedWalletId,
+                passphrase,
+                walletEnckey
+              ).subscribe((data) => {
                 this.setWalletViewKey(data["result"]);
-              }
-            )
-            this.checkWalletTxnHistory(selectedWalletId, passphrase).subscribe(
-              data => {
+              });
+              this.checkWalletTxnHistory(
+                selectedWalletId,
+                passphrase,
+                walletEnckey
+              ).subscribe((data) => {
                 this.setWalletTxnHistory(data["result"]);
-              }
-            );
-          }
-        });
+              });
+            }
+          });
+        }
       }
-    });
+    );
 
     return result;
   }
 
-  addWallet(id: string, passphrase: string): Observable<string> {
+  addWallet(
+    id: string,
+    passphrase: string,
+    mnemonics: string
+  ): Observable<string> {
+    localStorage.setItem(`${id}_passphrase`, passphrase);
+    localStorage.setItem(`${id}_mnemonics`, mnemonics);
```

Comment:
As warned by https://github.com/crypto-com/sample-chain-wallet/pull/57#issuecomment-609712638 -- storing sensitive information in localStorage isn't ideal

Reply:
ok, i'll remove it